code/game/objects/items/weapons/pneumaticCannon.dm

```diff
@@ -164,3 +211,28 @@
 		return
 	add_overlay(tank.icon_state)
 	src.update_icon()
+
+/obj/item/weapon/pneumatic_cannon/proc/fill_with_type(type, amount)
+	if(!ispath(type, /obj/item))
+		return FALSE
+	for(var/i in 1 to amount)
+		var/obj/item/I = new type
+		if(!load_item(I, null))
+			return TRUE
+
+/obj/item/weapon/pneumatic_cannon/pie
+	name = "pie cannon"
+	desc = "Load cream pie for optimal results"
+	force = 10
+	icon_state = "piecannon"
+	gasPerThrow = 0
+	checktank = FALSE
+	range_multiplier = 3
+	fire_mode = PCANNON_FIFO
+	throw_amount = 1
+	maxWeightClass = 100	//50 pies. :^)
+	clumsyCheck = FALSE
+
+/obj/item/weapon/pneumatic_cannon/pie/attackby(obj/item/I, mob/living/L)
+	if(istype(I, /obj/item/weapon/reagent_containers/food/snacks/pie))
+		return ..()
```

Comment:
Two refactors are needed add a `CanLoadItemProc` and remove this attackby.
ALSO the parent's attackby doesn't check for harm intent